### ISSUE TITLE
delete actors 

### ISSUE THREAD

**Issue by laureD19** [SSHOC/marketplace-curation#31]
Currently actors attached to items can't be deleted, even when setting the force parameter to `true` . DELETE /api/actors{id} returns a 400 with `"error": "Cannot delete actors that are contributors of active items!"`

List here actors that should nonetheless be deleted:
- /api/actors/2196
- api/actors/2886

**Comment by KlausIllmayer** [comment]
Indeed, very strange: the actor 2196 on production is not connected to an active item based on the database. We need to find out, what prevents the deletion of it. Maybe there is a draft connected to the actor? Will do further research.

**Comment by laureD19** [comment]
just checked again, and still running into the same issue. I found work-arounds (not ideal) to get rid of annoying actors I wanted to delete. But I still run into a 400 with, for example, on production:
- the `DELETE` /api/actors/2890

notify also @mkrzmr

**Comment by mkrzmr** [comment]
when trying to delete actor 2890:
get "Cannot delete actors that are contributors of items!"
try again with admin account and force = True, but still fails
search for all items by 2890 including drafts = lots of results

I know a workaround would be to edit all items, remove 2890 and then delete the actor. But I wonder why force=True with admin account does not delete the actor

**Comment by mkrzmr** [comment]
> Indeed, very strange: the actor 2196 on production is not connected to an active item based on the database. We need to find out, what prevents the deletion of it. Maybe there is a draft connected to the actor? Will do further research.

I used {{ _.host }}/api/actors/2196 with items=true and got some results. The items seem to be drafts? I could not find them through item-search

Edit: Looks like these items are old versions

**Comment by laureD19** [comment]
deletion also needed for:

- /api/actors/2842
- /api/actors/3798
- /api/actors/3790

**Comment by mkrzmr** [comment]
It's the same with these actors. If you use /api/actors/2196 with items=true you see they are connected to items (I suppose old versions) that do not seem to be affected by a force delete

**Comment by laureD19** [comment]
Yes, it is the same. Maybe Tomasz can include the exclusion of drafts cases in the force parameter??

Then, I also get another situation. When trying to delete, I get the following error code `"error": "Cannot delete actors that are affiliations of other actors!"`, although I can't see any affiliations when looking for them. 
- 8843
- 7332
- 11759
- 3627
- 4529
- 4100
- 4485

**Comment by KlausIllmayer** [comment]
> Then, I also get another situation. When trying to delete, I get the following error code `"error": "Cannot delete actors that are affiliations of other actors!"`, although I can't see any affiliations when looking for them.

Indeed, all of the eight mentions actors are affiliations of other actors. I can see this in the database but I'm not sure if we are able to see it by using API calls.

**Comment by KlausIllmayer** [comment]
I don't find an API call to identify if an actor is an affiliation of other actors - so we currently don't have a single API call to identify which actors belong to an affiliation actor id. This is relevant because for deleting the actor who is an affiliation of other actors these affiliations to the other actors must be deleted beforehand. I now see two options:

1.  force deleting an actor also deletes the affiliations
2. have an API call to identify the actors that have the affiliation of the concerned actor

I guess, we like to go with (1) - what do you think @laureD19 @mkrzmr - if we agree we can create an issue for backend.

**Comment by KlausIllmayer** [comment]
The reason why some actors can't be deleted are described here: https://github.com/SSHOC/sshoc-marketplace-backend/issues/468
